Makefile

```diff
@@ -5,25 +5,72 @@ REPO_PATH   ?= $(ORG_PATH)/$(PROJECT)
 VERSION     ?= $(shell git describe --always --dirty --abbrev=7)
 LD_FLAGS    ?= -X $(REPO_PATH)/pkg/version.Raw=$(VERSION)
 BUILD_DEST  ?= bin/vertical-pod-autoscaler-operator
+REGISTRY_SETUP_BINARY ?= bin/registry-setup
 MUTABLE_TAG ?= latest
 IMAGE        = origin-vertical-pod-autoscaler-operator
 
+# Add OUTPUT_DIR for various targets
+OUTPUT_DIR 						:= ./_output
+ARTIFACTS 						:= ./artifacts
+OLM_ARTIFACTS 					:= $(ARTIFACTS)/olm
+KUBE_MANIFESTS_DIR 				:= $(OUTPUT_DIR)/deployment
+OPERATOR_REGISTRY_MANIFESTS_DIR := $(OUTPUT_DIR)/olm/registry
+OLM_MANIFESTS_DIR 				:= $(OUTPUT_DIR)/olm/subscription
+
+KUBECTL = kubectl
+REGISTRY_VERSION	:= 4.8
+
+OPERATOR_NAMESPACE			:= openshift-vertical-pod-autoscaler
+OPERATOR_DEPLOYMENT_NAME	:= vertical-pod-autoscaler-operator
+
+export OLD_OPERATOR_IMAGE_URL_IN_CSV 	= quay.io/openshift/vertical-pod-autoscaler-operator:$(REGISTRY_VERSION)
+export OLD_OPERAND_IMAGE_URL_IN_CSV 	= quay.io/openshift/vertical-pod-autoscaler:$(REGISTRY_VERSION)
+export CSV_FILE_PATH_IN_REGISTRY_IMAGE 	= /manifests/$(REGISTRY_VERSION)/vertical-pod-autoscaler.v$(REGISTRY_VERSION).0.clusterserviceversion.yaml
+
+# Added LOCAL_OPERATOR_IMAGE for local-image build
+DEV_REPO			?= quay.io/redhat
+DEV_OPERATOR_IMAGE	?= openshift-vertical-pod-autoscaler-operator
+DEV_OPERAND_IMAGE	?= openshift-vertical-pod-autoscaler
+DEV_REGISTRY_IMAGE	?= vpa-operator-registry
+
+LOCAL_OPERATOR_IMAGE	?= $(DEV_REPO)/$(DEV_OPERATOR_IMAGE):latest
+LOCAL_OPERAND_IMAGE 	?= $(DEV_REPO)/$(DEV_OPERAND_IMAGE):latest
+LOCAL_OPERATOR_REGISTRY_IMAGE ?= $(DEV_REPO)/$(DEV_REGISTRY_IMAGE):latest
+export LOCAL_OPERATOR_IMAGE
+export LOCAL_OPERAND_IMAGE
+export LOCAL_OPERATOR_REGISTRY_IMAGE
+
 ifeq ($(DBG),1)
 GOGCFLAGS ?= -gcflags=all="-N -l"
 endif
 
+GO_BUILD_BINDIR := bin
+GO_TEST_PACKAGES :=./pkg/... ./cmd/...
+
 .PHONY: all
 all: build images check
 
 NO_DOCKER ?= 0
+HASDOCKER := $(shell command -v docker 2> /dev/null)
 ifeq ($(NO_DOCKER), 1)
   DOCKER_CMD =
   IMAGE_BUILD_CMD = imagebuilder
 else
-  DOCKER_CMD := podman run --rm -v "$(CURDIR):/go/src/$(REPO_PATH):Z" -w "/go/src/$(REPO_PATH)" openshift/origin-release:golang-1.15
-  IMAGE_BUILD_CMD = buildah bud
+	ifndef HASDOCKER
+	  DOCKER_CMD := podman run --rm -v "$(CURDIR):/go/src/$(REPO_PATH):Z" -w "/go/src/$(REPO_PATH)" openshift/origin-release:golang-1.12
+	  IMAGE_BUILD_CMD = buildah bud
+	  IMAGE_PUSH_CMD = podman push
+	else
+	  DOCKER_CMD := docker run --rm -v "$(CURDIR):/go/src/$(REPO_PATH):Z" -w "/go/src/$(REPO_PATH)" openshift/origin-release:golang-1.12
+	  IMAGE_BUILD_CMD = docker build
+	  IMAGE_PUSH_CMD = docker push
+	endif
 endif
 
+# build image for ci
+CI_REPO ?=registry.svc.ci.openshift.org
+$(call build-image,vertical-pod-autoscaler-operator,$(CI_IMAGE_REGISTRY)/autoscaling/vertical-pod-autoscaler-operator,./images/ci/Dockerfile,.)
```

Comment:
does this run the build-image target unconditionally? I'd prefer to associate this only with targets that need it